src/H5.c

```diff
@@ -88,6 +95,12 @@ H5_debug_t     H5_debug_g; /* debugging info */
 /* Local Variables */
 /*******************/
 
+/* Linked list of registered 'atclose' functions to invoke at library shutdown */
+static H5_atclose_node_t *H5_atclose_head = NULL;
+
+/* Declare a free list to manage the H5_atclose_node_t struct */
+H5FL_DEFINE_STATIC(H5_atclose_node_t);
```

Comment:
I feel like using a free list here is overkill. I seriously doubt there's going to be a lot of memory churn on close callbacks.

Reply:
also tend to agree with Dana